core-common/src/main/java/org/zalando/nakadi/exceptions/runtime/InvalidCursorException.java

```diff
@@ -9,29 +9,43 @@
     private final CursorError error;
     private final Cursor cursor;
     private final NakadiCursor position;
+    private final String eventType;
 
-    public InvalidCursorException(final CursorError error, final Cursor cursor) {
+    public InvalidCursorException(final CursorError error, final Cursor cursor, final String eventType) {
         this.error = error;
         this.cursor = cursor;
         this.position = null;
+        this.eventType = eventType;
     }
 
     public InvalidCursorException(final CursorError error, final NakadiCursor position) {
         this.error = error;
         this.cursor = null;
+        this.eventType = null;
         this.position = position;
     }
 
-    public InvalidCursorException(final CursorError error) {
+    public InvalidCursorException(final CursorError error, final String eventType) {
         this.error = error;
         this.cursor = null;
         this.position = null;
+        this.eventType = eventType;
     }
 
     public CursorError getError() {
         return error;
     }
 
+    private String getEventType() {
+        if (null != position) {
```

Comment:
is it possible the position is null ?

Reply:
Yes, it is possible. 
I decided to make it small, and therefore did in the same way as `#getPartition` and `#getOffset` of this class